### PR TITLE
Ensure Redis Enterprise Cluster tags can be set

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #13076

### CODE REVIEW

**Comment by cloutier** [comment]
I'm the one who broke it in a PR.

I see the mistake now. I was looking at the regular redis to understand how to do it and I actually based my PR on the vanilla redis data source instead by accident. My patch worked for me because I don't set tags myself, they are inherited. 

This fixes it the right way for everyone! Thanks

**Comment by stack72** [comment]
> I'm the one who broke it in a PR.
> 
> I see the mistake now. I was looking at the regular redis to understand how to do it and I actually based my PR on the vanilla redis data source instead by accident. My patch worked for me because I don't set tags myself, they are inherited.
> 
> This fixes it the right way for everyone! Thanks

It happens :) There are a lot of different ways Tags can be set - it was a very nice simple fix :)

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for this @stack72

File: internal/services/redisenterprise/redis_enterprise_cluster_resource.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func resourceRedisEnterpriseCluster() *pluginsdk.Resource {
 				Deprecated: "This field currently is not yet being returned from the service API, please see https://github.com/Azure/azure-sdk-for-go/issues/14420 for more information",
 			},
 
-			"tags": tags.SchemaDataSource(),
+			"tags": tags.Schema(),
```
Comment (tombuildsstuff) [inline internal/services/redisenterprise/redis_enterprise_cluster_resource.go:106]: actually this resource doesn't support update, so this has to be ForceNew unless we implement an Update function here:

>         	* resource azurerm_redis_enterprise_cluster: No Update defined, must set ForceNew on: []string{"tags"}
366

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @stack72 

Thanks again for this PR.

Taking a look through I'd missed the unit tests are failing because there's no Update function defined for this (and thus this field can't be ForceNew without an update method implemented) - so for the moment this'd either need an Update method implemented or to be made ForceNew, WDYT?

Thanks

**Comment by cloutier** [comment]
I can't seem to figure out from the logs which test broke though

**Comment by tombuildsstuff** [comment]
@cloutier it's the `TestProvider` test, which validates the schema's valid (e.g. in this case, a field can only be updated if there's an update function for that resource - otherwise all fields have to be ForceNew/Computed): https://github.com/hashicorp/terraform-provider-azurerm/pull/13078/checks?check_run_id=3381649396#step:5:363

File: internal/services/redisenterprise/redis_enterprise_cluster_resource.go (hunk 1 above)
Comment (stack72) [reply internal/services/redisenterprise/redis_enterprise_cluster_resource.go:106]: So the original PR should be reverted - right now, the resource is just broken as no tags can be set - I am happy to close this PR out in favour of moving back to the ForceNew - happy for you to make the call here

**Comment by cloutier** [comment]
Force new will break my use case! There are other processes in my org that updates tags regularely. My DB will be wiped everytime I deploy

File: internal/services/redisenterprise/redis_enterprise_cluster_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/redisenterprise/redis_enterprise_cluster_resource.go]: imho if the API supports update, we should implement an update function and correctly implement tags - if it doesn't then we have to use forcenew

Comment (cloutier) [reply internal/services/redisenterprise/redis_enterprise_cluster_resource.go]: I'm working on that right now, but I'm doing that to teach myself go. If anyone wants to go ahead they are welcome to!

Comment (cloutier) [reply internal/services/redisenterprise/redis_enterprise_cluster_resource.go]: I'm doing that here: https://github.com/hashicorp/terraform-provider-azurerm/pull/13084

**Comment by tombuildsstuff** [comment]
hey @stack72 

Thanks for this PR - since @cloutier has opened #13084 which adds an Update function, whilst I'd like to thank you for this contribution I'm going to close this in favour of #13084, which'll make these tags updateable.

Thanks!

**Pull request closed without merging**